test_runner/regress/test_download_extensions.py

```diff
@@ -113,40 +100,38 @@ def test_remote_extensions(
         remote_ext_config=env.ext_remote_storage.to_string(),
         # config_lines=["log_min_messages=debug3"],
     )
-    try:
-        with closing(endpoint.connect()) as conn:
-            with conn.cursor() as cur:
-                # Check that appropriate control files were downloaded
-                cur.execute("SELECT * FROM pg_available_extensions")
-                all_extensions = [x[0] for x in cur.fetchall()]
-                log.info(all_extensions)
-                assert "anon" in all_extensions
-
-                # postgis is on real s3 but not mock s3.
-                # it's kind of a big file, would rather not upload to github
-                if remote_storage_kind == RemoteStorageKind.REAL_S3:
-                    assert "postgis" in all_extensions
-                    # this may fail locally if dependency is missing
-                    # we don't really care about the error,
-                    # we just want to make sure it downloaded
-                    try:
-                        cur.execute("CREATE EXTENSION postgis")
-                    except Exception as err:
-                        log.info(f"(expected) error creating postgis extension: {err}")
-                        # we do not check the error, so this is basically a NO-OP
-                        # however checking the log you can make sure that it worked
-                        # and also get valuable information about how long loading the extension took
-
-                # this is expected to fail on my computer because I don't have the pgcrypto extension
+    with closing(endpoint.connect()) as conn:
+        with conn.cursor() as cur:
+            # Check that appropriate control files were downloaded
+            cur.execute("SELECT * FROM pg_available_extensions")
+            all_extensions = [x[0] for x in cur.fetchall()]
+            log.info(all_extensions)
+            assert "anon" in all_extensions
+
+            # postgis is on real s3 but not mock s3.
+            # it's kind of a big file, would rather not upload to github
+            if remote_storage_kind == RemoteStorageKind.REAL_S3:
+                assert "postgis" in all_extensions
+                # this may fail locally if dependency is missing
+                # we don't really care about the error,
+                # we just want to make sure it downloaded
                 try:
-                    cur.execute("CREATE EXTENSION anon")
+                    cur.execute("CREATE EXTENSION postgis")
                 except Exception as err:
```

Comment:
Use `with pytest.raises` instead of try..except (you can find examples in other tests).

Reply:
@lubennikovaav do you think we should re-write tests with extensions that we actually can download? 
I kind of feel like if all of our tests are wrapped in try-excepts some errors could slip through.

Reply:
also @bayandin I can't use `pytest.raises` because it's ok if the error is not raised
like, on my local machine the error will be raised because I'm missing the extension dependencies. but on github the error won't be raised because it has the dependenceis

Reply:
> also @bayandin I can't use `pytest.raises` because it's ok if the error is not raised like, on my local machine the error will be raised because I'm missing the extension dependencies. but on github the error won't be raised because it has the dependencies

This sounds like undefined behaviour. The test should be more deterministic

Reply:
> We should add a test extension (something really simple, almost a noop, maybe from the postgres repo itself) source code to the repo and build it in makefile. Then it'll work the same for local and CI runs (and on linux and macOS).

sounds good. I think I have done this